lib/user_agent/operating_systems.rb

```diff
@@ -18,42 +20,44 @@ module OperatingSystems
       "Windows CE"      => "Windows CE"
     }.freeze
 
-    def self.normalize_os(os)
-      Windows[os] || normalize_mac_os_x(os) || normalize_ios(os) || normalize_chrome_os(os) || os
+    class << self
+    def Windows # rubocop:disable Naming/MethodName
+      warn("#{__method__} is deprecated. Please use WINDOWS instead")
+      WINDOWS
     end
 
-    private
-      def self.normalize_chrome_os(os)
-        if os =~ /CrOS\s([^\s]+)\s(\d+(\.\d+)*)/
-          if $2.nil?
-            "ChromeOS"
-          else
-            version = $2
-            "ChromeOS #{version}"
-          end
-        end
+    def normalize_os(os)
+      WINDOWS[os] || normalize_mac_os_x(os) || normalize_ios(os) || normalize_chrome_os(os) || os
+    end
+
+      private
+
+    def normalize_chrome_os(os)
+      if os =~ CHROMEOS_REGEX && Regexp.last_match(2).nil?
+        "ChromeOS"
+      elsif os =~ CHROMEOS_REGEX
+        version = Regexp.last_match(2)
+        "ChromeOS #{version}"
       end
+    end
 
-      def self.normalize_ios(os)
-        if os =~ IOS_VERSION_REGEX
-          if $1.nil?
-            "iOS"
-          else
-            version = $1.tr('_', '.')
-            "iOS #{version}"
-          end
-        end
+    def normalize_ios(os)
+      if os =~ IOS_VERSION_REGEX && Regexp.last_match(1).nil?
+        "iOS"
+      elsif os =~ IOS_VERSION_REGEX
+        version = Regexp.last_match(1).tr("_", ".")
+        "iOS #{version}"
       end
+    end
 
-      def self.normalize_mac_os_x(os)
-        if os =~ /(?:Intel|PPC) Mac OS X\s*([0-9_\.]+)?/
-          if $1.nil?
-            "OS X"
-          else
-            version = $1.tr('_', '.')
-            "OS X #{version}"
-          end
-        end
+    def normalize_mac_os_x(os)
+      if os =~ MACOS_REGEX && Regexp.last_match(1).nil?
+        "OS X"
+      elsif os =~ MACOS_REGEX
+        version = Regexp.last_match(1).tr("_", ".")
+        "OS X #{version}"
       end
+    end
+    end
```

Comment:
Indentation is a bit weird in this class, starts at `class << self` I think. Surprised Rubocop didn't flag this!

Reply:
with how "verbose" it likes to be, i'm shocked too, but if rubocop is happy, i'm just happy it's not yelling